addon/queries/is.js

```diff
@@ -0,0 +1,56 @@
+import { findElementWithAssert, every } from '../helpers';
+
+/**
+ * Validates if an element (or elements) matches a given selector.
+ *
+ * Useful for checking if an element (or elements) matches a selector like
+ * `:disabled` or `:checked`, which can be the result of either an attribute
+ * (`disabled="disabled"`, `disabled=true`) or a property (`disabled`).
+ *
+ * @example
+ * // <input type="checkbox" checked="checked">
+ * // <input type="checkbox" checked>
+ *
+ * const page = PageObject.create({
+ *   areInputsChecked: is(':checked', 'input', { multiple: true })
+ * });
+ *
+ * assert.equal(page.areInputsChecked, true, 'Inputs are checked');
+ *
+ * @example
+ * // <button class="toggle-button active" disabled>Toggle something</button>
+ *
+ * const page = PageObject.create({
+ *   isToggleButtonActive: is('.active:disabled', '.toggle-button')
+ * });
+ *
+ * assert.equal(page.isToggleButtonActive, true, 'Button is active');
+ *
+ * @param {string} testSelector - CSS selector to test
+ * @param {string} targetSelector - CSS selector of the element to check
+ * @param {Object} options - Additional options
+ * @param {string} options.scope - Nests provided scope within parent's scope
+ * @param {boolean} options.resetScope - Override parent's scope
+ * @param {number} options.at - Reduce the set of matched elements to the one at the specified index
+ * @param {boolean} options.multiple - If set, the function will return an array of values
+ * @param {String} options.testContainer - Context where to search elements in the DOM
+ * @return {Descriptor}
+ *
+ * @throws Will throw an error if no element matches selector
+ * @throws Will throw an error if multiple elements are matched by selector and multiple option is not set
+ */
+export function is(testSelector, targetSelector, options = {}) {
+  return {
+    isDescriptor: true,
+
+    get() {
+      const elements = findElementWithAssert(this, targetSelector, options);
```

Comment:
We need to update this definition to support the new better errors message

```js
get(key) {
    const elements = findElementWithAssert(this, targetSelector, { ...options, pageObjectKey: key });
    ...
```

See https://github.com/san650/ember-cli-page-object/blob/948aec01fe0fd9a94438c63a5380b9d02a8a1ca5/addon/queries/text.js#L89-L90